### PR TITLE
Add SplitOuter, enhance Conv2d schedule and fix bugs

### PR DESCRIPTION
**Pull request by haozech**
This PR did 5 things:

1. Add schedule primitive SplitOuter and related unittest
2. Enhance schedule primitive SyncThreads
3. Enhance editing temp tensor's shape and buffer's size
4. Add conv2d schedule which is identical to tvm and related unittest
5. Fix bugs and improve readability

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -129,15 +129,16 @@ function build {
...
-    make test01_elementwise_add_main -j $JOBS
-    make test02_matmul_main -j $JOBS
-    make test03_conv_main -j $JOBS
-    make test_codegen_c -j $JOBS
     if [[ $cuda_config == "ON" ]]; then
         make test_codegen_cuda_dev -j $JOBS
         ctest -R test_codegen_cuda_dev -V
     fi
 
+    make test01_elementwise_add_main -j $JOBS
```
Comment (wenming2014) [inline build.sh:137]: 为什么要改这里？

File: cinn/poly/stage.h [hunk 2]
```diff
@@ -159,6 +159,21 @@ class Stage : public Object {
...
 
+  /**
+   * Split the loop level of into two new loop levels.
+   * @param level the level to split.
+   * @param nparts the extent(size) of the outer loop created after splitting.
+   * @return the new outer and inner iterators.
+   */
+  // @{
+  std::tuple<Iterator, Iterator>  //
+  SplitOuter(const Iterator& level, int nparts);
```
Comment (wenming2014) [inline cinn/poly/stage.h:170]: 和split的区别是？

File: tools/paddle_benchmark/paddle_test_benchmark.py [hunk 3]
```diff
@@ -25,8 +26,13 @@ def main():
 
     for _ in range(0, 10):
         predictor.zero_copy_run()
+
+    time1 = time.time()
     for i in range(0, 500):
         predictor.zero_copy_run()
+    time2 = time.time()
+    total_inference_cost = (time2 - time1) * 1000  # 总时延，ms
```
Comment (wenming2014) [inline tools/paddle_benchmark/paddle_test_benchmark.py]: 尽量不要带中文

File: build.sh (hunk 1 above)
Comment (haozech) [reply build.sh:137]: 将Cuda端测试提前，测试检查问题时更方便些

File: cinn/poly/stage.h (hunk 2 above)
Comment (haozech) [reply cinn/poly/stage.h:170]: 假如j的range是0:9，那么Split(j,2)得到的循环是：
for j_outer = 0:5
  for j_inner = 0:2

SplitOuter(j,2)得到的循环是：
for j_outer = 0:2
  for j_inner = 0:5

添加此函数主要是为了对齐TVM中split的使用方法，方便之后schedule开发

File: tools/paddle_benchmark/paddle_test_benchmark.py (hunk 3 above)
Comment (haozech) [reply tools/paddle_benchmark/paddle_test_benchmark.py]: 已修改 感谢

**Review by wenming2014 — APPROVED** [review]

**Pull request merged**